src/slider/index.ts

```diff
@@ -0,0 +1,172 @@
+  }
+
+  moveHandle(handle: Element, value: number) {
+    this.setup(handle);
+    this.moveElements(this.percentage(value - this.props.min, this.props.max - this.props.min));
+  }
+
+  onClick = (event: MouseEvent & Slider.Event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    const { handleLower, handleUpper } = this.state;
+    const handleLowerLeft = handleLower.getBoundingClientRect().left;
+    const handleUpperRight = handleUpper.getBoundingClientRect().right;
+    const mouseLeft = event.clientX;
+    if (Math.abs(mouseLeft - handleLowerLeft) < Math.abs(mouseLeft - handleUpperRight)) {
+      this.setup(handleLower);
+    } else {
+      this.setup(handleUpper);
+    }
+    this.updateFromPercentage(mouseLeft);
+  }
+
+  onDragStart = (event: MouseEvent & Slider.Event | TouchEvent & Slider.Event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.setup(event.target);
+    utils.WINDOW().document.addEventListener('mousemove', this.onMouseMove);
+    utils.WINDOW().document.addEventListener('touchmove', this.onTouchMove);
+    utils.WINDOW().document.addEventListener('mouseup', this.onDragEnd);
+    utils.WINDOW().document.addEventListener('touchend', this.onDragEnd);
+  }
+
+  onMouseMove = (event: MouseEvent & Slider.Event) => {
+    this.updateFromPercentage(event.clientX);
+  }
+
+  onTouchMove = (event: TouchEvent & Slider.Event) => {
+    const touchEvent = event.changedTouches[0];
+    if (touchEvent.target === this.state.handle) {
+      this.updateFromPercentage(touchEvent.clientX);
+    }
+  }
+
+  onDragEnd = (event: MouseEvent & Slider.Event) => {
+    utils.WINDOW().document.removeEventListener('mousemove', this.onMouseMove);
+    utils.WINDOW().document.removeEventListener('touchmove', this.onTouchMove);
+    utils.WINDOW().document.removeEventListener('mouseup', this.onDragEnd);
+    utils.WINDOW().document.removeEventListener('touchend', this.onDragEnd);
+  }
+
+  setup = (handle: Element) => {
+    const baseDimensions = this.state.base.getBoundingClientRect();
+    const { handleLower, handleUpper } = this.state;
+    const isLower = handleLower === handle;
+    const limit = parseFloat((isLower ? handleUpper : handleLower).style.left);
+    this.state = {
+      ...this.state,
+      baseLength: baseDimensions.width,
+      baseLeft: baseDimensions.left,
+      handle,
+      isLower,
+      limit
+    };
+  }
+
+  // TODO: Had a lot of trouble naming this...
+  updateFromPercentage(mouseLeft: number) {
+    this.state = { ...this.state, mouseLeft };
+    const percentage = this.calcMouseToPercentage();
+    this.moveElements(percentage);
+    this.setInput(percentage);
+  }
+
+  moveElements(percentage: number) {
+    const overlap = percentage === this.state.limit;
+    this.state.handle.style.left = `${percentage}%`;
+    this.moveConnect(percentage);
+    if (overlap && this.state.isLower) {
+      this.state.handleLower.style['z-index'] = ZINDEX_HIGH;
```

Comment:
maybe `this.state.handleLower.style['z-index'] = overlap && this.state.isLower ? ZINDEX_HIGH : ZINDEX_LOW;`